Jain-sip change DialogState and TransactionState to enum type.

xhoaluu opened the pull request:
@jaimecasero 
I have update the code to change [DialogState](https://github.com/RestComm/jain-sip/blob/master/src/javax/sip/DialogState.java) and [TxState](https://github.com/RestComm/jain-sip/blob/master/src/javax/sip/TransactionState.java) to enum. cc-buildloop is green. Could you go through the code changes and give me comment.

jaimecasero commented:
@xhoaluu  can we try to respect the previous public API? For exmaple, you could add an getInternalState in new Enum,so current clients dont need to change...

xhoaluu commented:
@jaimecasero 
Just updated the code based on your comments. please take a look again.

Fixed:
Change DialogState and TransactionState to enum and still respect the current logic of code, dont change public API and dont delete any API.
use the constant names to again prevent as much as possible.

jaimecasero commented on src/javax/sip/TransactionState.java:
```diff
@@ -214,286 +214,47 @@
+    TERMINATED(5);
     
+    // Internal Variable
+    private int intValue = -1;
```
I would prefer not to have these int values... I was thinking about calling "PROCEEDING"  like the previous constant "_PROCEEDING" so client dont need to change...

xhoaluu replied:
@jaimecasero 
"I was thinking about calling "PROCEEDING" like the previous constant "_PROCEEDING" so client dont need to change..."

in the code we have a lot of code in src and tck where using PROCEEDING for compare about object, but when they need to compare int value, they use _PROCEDDING. In my view, if we use constant name _PROCEEDING, then we also have a lot of change in code.

for int value, Do you mean that I need to use ordinal() value of enum to do compare?

jaimecasero commented:
This was original constant https://github.com/RestComm/jain-sip/blob/master/src/javax/sip/TransactionState.java#L436 , so i was thinking about keeping the same identifier "_PROCEEDDING" for the new enum so client code still refers to valid stuff, but in that case enum value

xhoaluu commented:
@jaimecasero 
"Notice on Java enum values maybe compared both with equals and "==" operator. So, in the end an enum value could work potentially as an "int""

enum.quals(int) does not work at all even the compiler can build the source. Could you specify more detail.

jaimecasero commented:
@xhoaluu  I noticed these classes are part of JAINSIP spec, so we are not really allowed to change these classes. Lets pause this ticket by now,ok? I will find something elese for you in the meantime
